src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -1870,7 +1870,7 @@ bool WindowImplX11::processEvent(XEvent& windowEvent)
                     static XComposeStatus status;
                     std::array<char, 16>  keyBuffer{};
                     if (XLookupString(&windowEvent.xkey, keyBuffer.data(), keyBuffer.size(), nullptr, &status))
-                        pushEvent(Event::TextEntered{static_cast<std::uint32_t>(keyBuffer[0])});
+                        pushEvent(Event::TextEntered{static_cast<char32_t>(keyBuffer[0])});
```

Comment:
Kind of wonder if this is correct. Not relevant for this PR, but assuming this return UTF-8 (only thing fitting `char`), I guess we'd need to actually convert it like we do for the other implementations, instead of just casting it.

Reply:
I assume that static casting to character types maintains the integer value of the character so if that's what we were doing before, I suspect that's still the correct thing to do now.

Reply:
With "Not relevant for this PR" I referred to the current implementation and not to the change.

Since I assume Linux returns UTF-8 `char` and we just do a cast to `char32_t` and represent it as UTF-32 character, I doubt that his is correct. (Completely unrelated whether it's `std::uint32_t` or `char32_t`.)

Reply:
Oh I see what you mean now. My bad.